parser/src/format/character_separated/mod.rs

```diff
@@ -406,6 +422,66 @@ mod test {
         Input::Stream(Box::new(Cursor::new(content.into())))
     }
 
+    // Test for: https://github.com/estuary/connectors/issues/97
+    #[test]
+    fn fails_when_there_are_more_columns_than_headers() {
+        let conf = ParseConfig::default();
+
+        // CSV has 2 column headers, but row 3 has 3 columns :boom:
+        let csv = test_input("a,b\n1\n2,3\n4,5,6");
```

Comment:
Maybe split this CSV input into a multi-line string for readability?

Reply:
Generally for inline CSVs I've preferred to leave it as a single line because multiline rust strings don't automatically strip indentation. There's a crate that has a procedural macro for it, but I think my preference would be to have any test cases that are significantly larger than this one live as separate files and use `include_bytes!`.